src/Analysis/Elaboration.hs

```diff
@@ -24,69 +25,81 @@ runElab :: Elab a -> (a, Subst PartialType)
 runElab = fst . flip runFresh (Name 0) . flip runReaderT mempty . flip runStateT mempty
 
 elaborate :: Term -> Elab PartialElabTerm
-elaborate (Fix (Abs n b)) = do
+elaborate term = do
+  term' <- infer term
+  subst <- get
+  let ty :< tm = substitute subst term'
```

Comment:
We should probably be applying the substitution a little more eagerly to avoid making redundant passes through terms and their types, but in the meantime this is a decent way to handle it as a high-level entry point.